### PR TITLE
Adding ADPICam Detector and Cam classes

### PR DESCRIPTION
**Pull request by jwlodek**
This PR adds Detector and Cam classes for the [ADPICam](https://github.com/areaDetector/ADPICam) driver for Princeton Instruments detectors. It was mostly created using the boilerplating script from PR #975, with minor touchups after to fix some attribute names.

ADPICam has a massive number of unique PV names, since there is a binary Yes/No PV for each camera feature to check if the feature exists, and also to check if it enabled (PVs with _EX and _PR respectively). I wasn't sure if these should remain included or not.

The file this was all pulled from was from the driver:
https://github.com/areaDetector/ADPICam/blob/master/PICamApp/Db/PICam.template

### CODE REVIEW

**Review by mrakitin — APPROVED** [review]
Looks good to me! It's a nice demonstration of the usefulness of #975.

File: ophyd/areadetector/cam.py [hunk 1]
```diff
@@ -586,6 +587,388 @@ class PSLDetectorCam(CamBase):
...
+    clean_section_final_height_count = ADCpt(SignalWithRBV, 'CleanSectionFinalHeightCount')
+    clean_serial_register = ADCpt(SignalWithRBV, 'CleanSerialRegister')
+    clean_until_trigger = ADCpt(SignalWithRBV, 'CleanUntilTrigger')
+    disable_cooling_fan = ADCpt(SignalWithRBV, 'DisableCoolingFan')
+    enable_sensor_window_heater = ADCpt(SignalWithRBV, 'EnableSensorWindowHeater')
+    sensor_temperature_status = ADCpt(EpicsSignal, 'SensorTemperatureStatus')
+    enable_roi_min_x_input = ADCpt(EpicsSignal, 'EnableROIMinXInput')
+    enable_roi_size_x_input = ADCpt(EpicsSignal, 'EnableROISizeXInput')
+    enable_roi_min_y_input = ADCpt(EpicsSignal, 'EnableROIMinYInput')
+    enable_roi_size_y_input = ADCpt(EpicsSignal, 'EnableROISizeYInput')
```
[the remark this answers is not in the document]
Comment (jwlodek) [reply ophyd/areadetector/cam.py:969]: From the ADPICam documentation:

```
_EX: Parameter Exists indicators

_PR: Parameter Relevence indicators
```

I think the `_EX` is just a PV saying if the current device supports that specific feature, and `_PR` says if the feature is enabled in the current operating mode

[the remark this answers is not in the document]
Comment (mrakitin) [reply ophyd/areadetector/cam.py:969]: Got it, thanks.

**Comment by mrakitin** [comment]
@jwlodek, want to try to "rerender" it with the `inflection`-based script? If it produces the same output, I am happy to merge.

**Comment by jwlodek** [comment]
Seems to have everything the same, aside from this PV: `EMIccdGain_PR`, which `inflection` makes `em_iccd_gain_pr`, instead of `emi_ccd_gain_pr`, because the PV name doesn't capitalize Ccd.

**Comment by mrakitin** [comment]
> Seems to have everything the same, aside from this PV: `EMIccdGain_PR`, which `inflection` makes `em_iccd_gain_pr`, instead of `emi_ccd_gain_pr`, because the PV name doesn't capitalize Ccd.

Right, that's not a problem with `inflection`, but rather an issue with proper formatting of the PV name.

**Pull request merged**